android/app/src/main/java/io/logto/demo/viewmodel/LogtoViewModel.kt

```diff
@@ -0,0 +1,95 @@
+package io.logto.demo.viewmodel
+
+import android.app.Application
+import android.content.Context
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import io.logto.android.Logto
+import io.logto.client.config.LogtoConfig
+import io.logto.client.constant.ScopeValue
+import io.logto.client.exception.LogtoException
+import org.jose4j.jwt.JwtClaims
+
+class LogtoViewModel(application: Application) : AndroidViewModel(application) {
+
+    private lateinit var logto: Logto
+
+    private val _authenticated = MutableLiveData(false)
+    val authenticated: LiveData<Boolean>
+        get() = _authenticated
+
+    private val _idTokenClaims = MutableLiveData<JwtClaims>()
+    val idTokenClaims: LiveData<JwtClaims>
+        get() = _idTokenClaims
+
+    private val _accessToken = MutableLiveData<String>()
+    val accessToken: LiveData<String>
+        get() = _accessToken
+
+    private val _logtoException = MutableLiveData<LogtoException>()
+    val logtoException: LiveData<LogtoException>
+        get() = _logtoException
+
+    fun signIn(context: Context) {
+        logto.signInWithBrowser(context) { exception, tokenSet ->
+            if (exception !== null) {
+                _logtoException.postValue(exception)
+            } else {
+                _authenticated.postValue(logto.authenticated)
+            }
+        }
+    }
+
+    fun signOut(context: Context) {
+        logto.signOutWithBrowser(context) { exception ->
+            if (exception != null) {
+                _logtoException.postValue(exception)
+            }
+            _authenticated.postValue(logto.authenticated)
```

Comment:
should logout ahead

Reply:
Todo: [Align sign out process with js-sdk (LOG-265)](https://linear.app/silverhand/issue/LOG-265/align-sign-out-process-with-js-sdk)